File: algos/solutions/a1twopointers/JS/palindrome/algo.js
// algo.js
function isPalindrome(inputString) {
    const sanitized = inputString.toLowerCase().replace(/[^a-z0-9]/g, '');

    let left = 0, right = sanitized.length - 1;

    while (left < right) {
        if (sanitized[left] !== sanitized[right]) {
            return false;
        }

        left++;
        right--;
    }

    return true;
}

module.exports = { isPalindrome };

File: algos/solutions/a1twopointers/Go/palindrome/algo.go
package palindrome

import (
	"strings"
	"unicode"
)

// IsPalindrome checks if the input string is a palindrome after sanitizing.
func IsPalindrome(s string) bool {
	var sanitized strings.Builder
	for _, r := range s {
		if unicode.IsLetter(r) || unicode.IsDigit(r) {
			sanitized.WriteRune(unicode.ToLower(r))
		}
	}
	str := sanitized.String()
	left, right := 0, len(str)-1
	for left < right {
		if str[left] != str[right] {
			return false
		}
		left++
		right--
	}
	return true
}
